test/deploy/rackhd_microservice_install.py

```diff
@@ -0,0 +1,185 @@
+
+        # Make an nfs mount point
+        self.assertEqual(fit_common.remote_shell('sudo mkdir -p /nfs'
+                                                 )['exitcode'], 0, "Failed to mkdir /nfs")
+        # Ubuntu 14.04 nfs services
+        self.assertEqual(fit_common.remote_shell('sudo service nfs-kernel-server start'
+                                                 )['exitcode'], 0, "Failed to start nfs-kernel-server")
+        # export file system
+        self.assertEqual(fit_common.remote_shell('sudo exportfs -a'
+                                                 )['exitcode'], 0, "Failed on command exportfs -a")
+        # check mount point
+        self.assertEqual(fit_common.remote_shell('sudo showmount -e 172.31.128.1'
+                                                 )['exitcode'], 0, "Display exported fs")
+
+    def test03_install_docker(self):
+        # Check if docker-compose is already installed
+        rsp = fit_common.remote_shell("docker -v")
+        if rsp['exitcode'] == 0:
+            logs.info(" Docker already installed")
+            if "Docker version" in rsp['stdout']:
+                logs.info(" Docker installed: %s", rsp['stdout'])
+        else:
+            logs.info(" Install Docker")
+
+            # apt-get update
+            command = PROXYVARS + "sudo apt-get update"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "get-update failed.")
+
+            # Getting the docker installation file from docker.com
+            command = PROXYVARS + "sudo wget -qO- https://get.docker.com/ | sh"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Getting docker file from docker.com failed.")
+
+            # Checking the users with whoami command
+            command = "sudo usermod -aG docker $(whoami)"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "whoami command failed.")
+
+        # Check if docker-compose is already installed
+        rsp = fit_common.remote_shell("docker-compose -v")
+        if rsp['exitcode'] == 0:
+            logs.info(" docker-compose already installed")
+            if "docker-compose version" in rsp['stdout']:
+                logs.info(" Docker installed: %s", rsp['stdout'])
+        else:
+            # Installing python pip
+            command = PROXYVARS + "sudo apt-get -y install python-pip"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Pip install failed.")
+
+            # Installing docker compose
+            logs.info(" Install docker-compose")
+            command = PROXYVARS + "sudo pip install docker-compose"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Docker compose install failed.")
+
+        # Setup docker proxy environment
+        logs.info(" Setup Docker Proxy")
+        dockerproxy = open("dockerproxy", 'w')
+        dockerproxy.write('export http_proxy="http://web.hwimo.lab.emc.com:3128/"\n')
```

Comment:
We should probably not hard-code this, this can be pulled from fitcfg() install_default.json proxy variables.  If no proxy values defined, skip setting up docker proxy and give a warning.